### PR TITLE
raft: allow skip bcast commit

### PR DESCRIPTION
**Pull request by BusyJay**
ref coreos/etcd#8190

### CODE REVIEW

File: src/raft/raft.rs [hunk 1]
```diff
@@ -1624,7 +1633,7 @@ impl<T: Storage> Raft<T> {
 
         // The quorum size is now smaller, so see if any pending entries can
         // be committed.
-        if self.maybe_commit() {
+        if self.maybe_commit() && !self.skip_bcast_commit {
```
Comment (siddontang) [inline src/raft/raft.rs]: need check here?

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/raft/raft.rs (hunk 1 above)
Comment (BusyJay) [inline src/raft/raft.rs]: This doesn't happen frequently, whether check it doesn't make much difference. But as configuration name says, when it's enabled, broadcast of commit should be skipped.

**Comment by javaforfun** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Comment by hhkbp2** [comment]
It's better to make it into the etcd/raft before merging into tikv.

**Comment by hhkbp2** [comment]
Rest LGTM

File: src/raft/raft.rs (hunk 1 above)
Comment (ngaut) [inline src/raft/raft.rs]: I prefer to move check skip_bcast_commit flag before maybe_commit check.

Comment (BusyJay) [inline src/raft/raft.rs]: No, commit index should be updated no matter we want to bcast it or not, which will make all following heartbeats and append messages contain the latest commit index.

**Pull request merged**